api/controllers/service_api/wraps.py

```diff
@@ -172,6 +174,23 @@ def decorated(*args, **kwargs):
     return decorator
 
 
+@cache.memoize(timeout=3600)
+def get_api_token_from_db(auth_token, scope):
+    return db.session.query(ApiToken).filter(ApiToken.token == auth_token, ApiToken.type == scope).first()
+
+
+@shared_task(queue="ops_trace", bind=True, max_retries=3)
+def update_token_last_used_at(self, token_id, last_used_time):
+    try:
+        with db.session.begin():
+            token = db.session.query(ApiToken).get(token_id)
+            if token:
+                token.last_used_at = last_used_time
+    except Exception as e:
+        # 如果失败，最多重试3次
```

Comment:
Please make sure the comments should be in English

Reply:
sry, correct it